GlobalRegistrar/contract.sol

```diff
@@ -1,25 +1,65 @@
 //sol
 
-import "owned";
-
 contract NameRegister {
-	function addr(bytes32 _name) constant returns (address o_owner) {}
-	function name(address _owner) constant returns (bytes32 o_name) {}
+	function addr(bytes32 _name) constant returns (address o_owner);
+	function name(address _owner) constant returns (bytes32 o_name);
 }
 
 contract Registrar is NameRegister {
 	event Changed(bytes32 indexed name);
 	event PrimaryChanged(bytes32 indexed name, address indexed addr);
 
-	function owner(bytes32 _name) constant returns (address o_owner) {}
-	function addr(bytes32 _name) constant returns (address o_address) {}
-	function subRegistrar(bytes32 _name) constant returns (address o_subRegistrar) {}
-	function content(bytes32 _name) constant returns (bytes32 o_content) {}
+	function owner(bytes32 _name) constant returns (address o_owner);
+	function addr(bytes32 _name) constant returns (address o_address);
```

Comment:
This is confusing. You have defined "function addr(bytes32 _name) constant returns (address o_owner);" in contract NameRegister, and are here redefining it with the exact same interface but different return value. Can you explain that to me?